CRM/Contribute/Form/AdditionalPayment.php

```diff
@@ -299,8 +299,13 @@ public static function formRule($fields, $files, $self) {
     if (!empty($fields['net_amount']) && $netAmt != $fields['net_amount']) {
       $errors['net_amount'] = ts('Net amount should be equal to the difference between payment amount and fee amount.');
     }
-    if ($self->_paymentProcessor['id'] === 0 && empty($fields['payment_instrument_id'])) {
-      $errors['payment_instrument_id'] = ts('Payment method is a required field');
+    if (empty($fields['payment_instrument_id'])) {
+      if ($self->_paymentProcessor['id'] === 0) {
+        $errors['payment_instrument_id'] = ts('Payment method is a required field');
+      }
+    }
+    elseif ('Accounts Receivable' == CRM_Financial_BAO_FinancialTypeAccount::getInstrumentFinancialAccountName($fields['payment_instrument_id'])) {
```

Comment:
I've realized this is neither a sufficient nor necessary check. The Account Name is irrelevant. What needs to be tested is the Account Relationship between the financial account and any related financial type needs to be checked to see if it is 'is Accounts Receivable account'.

Reply:
@JoeMurray actually the from_account_id is derived from payment_instrument's linked FA (Financial Account), not financial type when we do a completed payment either from 'Record Payment' or backoffice contribution form. And there is no validation check imposed at form or BAO level to avoid such payment whose PI is linked with account receivable FA. Alternatively, will it be ok to test the Account Relationship of the PI (not FT) if its 'Accounts Receivable account is' or not, instead of FA name?

Reply:
@JoeMurray when I tried to check with Financial Account relationship (FAR) it comes to be 'Asset Account is' with accounting code AR. This is the same FAR for Deposit Bank Account, Payment Processor account etc. (not accounting code). So FAR can't be used to distinguish such custom PI instead we can use Accounting Code = AR which can't be changed by admin and the check would be:
 ```
elseif ('AR' == CRM_Financial_BAO_FinancialTypeAccount::getInstrumentFinancialAccountCode($fields['payment_instrument_id'])){
 ...
} 
```

Reply:
Documenting here for others to understand. So @monishdeb and discussed this a while ago: we're trying to see if, for the financial account used by the payment method, there are any account relationships between a financial type and this financial account which are 'Accounts Receivable Account is'.

Reply:
Even thought the check is only one line putting it in a function like $isAccountsReceiveablePayment (or whatever) will likely provide some clarity